File: leetcode/algorithms/482_license_key_formatting/main.py
class LicenseKeyFormatting:
    def licenseKeyFormatting(self, s: str, k: int) -> str:
        s = s.replace("-", "").upper()
        result = ""

        for i in range(len(s) - 1, -1, -k):
            if i >= k:
                result = "-" + s[-k:] + result
            else:
                result = s[-k:] + result
            s = s[: i + 1 - k]

        return result


    # Solution
    def solution(self, s: str, k: int) -> str:
        s = s.replace("-", "").upper()[::-1]
        return "-".join(s[i : i + k] for i in range(0, len(s), k))[::-1]

File: leetcode/algorithms/482_license_key_formatting/main.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var licenseKeyFormatting = function (s, k) {
  s = s.replaceAll("-", "");
  result = "";

  for (let i = s.length; i >= 0; i -= k) {
    if (i > k) {
      result = "-" + s.substring(i - k, i) + result;
    } else {
      result = s.substring(i - k, i) + result;
    }
  }

  return result.toUpperCase();
};



// Solution
var solution = function (s, k) {
  // Remove existing dashes and convert any lowercase letters to uppercase
  const newStr = s.replace(/-/g, "").toUpperCase();

  // Convert string to an array so we can manipulate it
  const arr = newStr.split("");

  for (let i = arr.length - 1 - k; i >= 0; i -= k) {
    // Loop through array backwards and decrement by value of K
    arr[i] = arr[i] + "-";
  }

  return arr.join("");
};
